tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1075 @@
+    "                to_dim,\n",
+    "                latent_dim,\n",
+    "                num_heads,\n",
+    "            ),\n",
+    "            TransformerDecoderTensorDict(\n",
+    "                num_blocks,\n",
+    "                from_name,\n",
+    "                to_name,\n",
+    "                from_dim,\n",
+    "                from_len,\n",
+    "                to_dim,\n",
+    "                latent_dim,\n",
+    "                num_heads,\n",
+    "            ),\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "15b1b4e2-918d-40bc-a245-15be0e9cc276",
+   "metadata": {},
+   "source": [
+    "We now test our new `TransformerTensorDict`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "7a680452-1462-4ee6-ba04-dce0bb855870",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "TensorDict(\n",
+       "    fields={\n",
+       "        Attn: Tensor(torch.Size([8, 2, 10, 3]), dtype=torch.float32),\n",
+       "        K: Tensor(torch.Size([8, 2, 3, 5]), dtype=torch.float32),\n",
+       "        Q: Tensor(torch.Size([8, 2, 10, 5]), dtype=torch.float32),\n",
+       "        V: Tensor(torch.Size([8, 2, 3, 5]), dtype=torch.float32),\n",
+       "        X_decode: Tensor(torch.Size([8, 10, 6]), dtype=torch.float32),\n",
+       "        X_encode: Tensor(torch.Size([8, 3, 5]), dtype=torch.float32),\n",
+       "        X_out: Tensor(torch.Size([8, 10, 6]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([8]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "to_dim = 5\n",
+    "from_dim = 6\n",
+    "latent_dim = 10\n",
+    "to_len = 3\n",
+    "from_len = 10\n",
+    "batch_size = 8\n",
+    "num_heads = 2\n",
+    "\n",
+    "tokens = TensorDict(\n",
+    "    {\n",
+    "        \"X_encode\": torch.randn(batch_size, to_len, to_dim),\n",
+    "        \"X_decode\": torch.randn(batch_size, from_len, from_dim),\n",
+    "    },\n",
+    "    batch_size=[batch_size],\n",
+    ")\n",
+    "\n",
+    "transformer = TransformerTensorDict(\n",
+    "    6,\n",
+    "    \"X_encode\",\n",
+    "    \"X_decode\",\n",
+    "    to_dim,\n",
+    "    to_len,\n",
+    "    from_dim,\n",
+    "    latent_dim,\n",
+    "    num_heads,\n",
+    ")\n",
+    "\n",
+    "transformer(tokens)\n",
+    "tokens"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3f6448dd-5d0d-43fd-9e57-a0ac3b30ecba",
+   "metadata": {},
+   "source": [
+    "We've achieved to create a transformer with `TensorDictModule`. This shows that `TensorDictModule`is a flexible module that can implement complex operarations"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd08362a-8bb8-49fb-8038-1a60c5c01ea2",
+   "metadata": {},
+   "source": [
+    "Have fun with TensorDictModule!"
```

Comment:
I wish we could have a performance measure of what it takes to run this transformer against a regular one, to see the overhead of tensordictmodule. Would it be possible to do that?